sklearn/utils/class_weight.py

```diff
@@ -64,35 +73,45 @@ def compute_class_weight(class_weight, *, classes, y, sample_weight=None):
     # Import error caused by circular imports.
     from sklearn.preprocessing import LabelEncoder
 
-    if set(y) - set(classes):
+    xp, _, device_ = get_namespace_and_device(y, classes)
+    max_float_dtype = _max_precision_float_dtype(xp, device=device_)
```

Comment:
I have the feeling that the `compute_class_weight` function should take `dtype`, `xp` and `device` as an argument to let the caller state the required precision level based on other things (such as the dtype of `X`).

Reply:
Other non-linear classifiers such as `MLPClassifier` probably behave similarly.

For others, like `HistGradientBoostingClassifier`, the numerical precision used to compute the weighted loss function and its gradient, is neither dependent on `X` nor on `y`.

Reply:
I just removed the `_max_precision_float_dtype` from this function and let the arrays be declared automatically on the device. We convert to the desired dtype anyway from where we are calling this.